src/zenml/integrations/huggingface/model_registries/huggingface_model_registry.py

```diff
@@ -0,0 +1,668 @@
+            repo_id=self._repo_id(name),
+            repo_type=self.REPO_TYPE,
+            discussion_num=pr.num,
+            new_status="open",
+        )
+
+        return ModelVersion(
+            version=pr_ref,
+            model_source_uri=pr.url,
+            model_format="hugginface",
+            registered_model=self.get_model(name),
+            description=description,
+            metadata=metadata,
+        )
+
+    def delete_model_version(
+        self,
+        name: str,
+        version: str,
+    ) -> None:
+        """Deletes a model version from the model registry.
+
+        Args:
+            name: The name of the registered model.
+            version: The version of the model version to delete.
+        """
+        self._model_exists(name)
+
+        for discussion in self.api.get_repo_discussions(
+            repo_id=self._repo_id(name), repo_type=self.REPO_TYPE
+        ):
+            if discussion.is_pull_request and discussion.url.endswith(
+                f"discussions/{version.split('/')[-1]}"
+            ):
+                self.api.change_discussion_status(
+                    repo_id=self._repo_id(name),
+                    repo_type=self.REPO_TYPE,
+                    discussion_num=discussion.num,
+                    new_status="closed",
+                )
+                break
+
+    def update_model_version(
+        self,
+        name: str,
+        version: str,
+        description: Optional[str] = None,
+        metadata: Optional[ModelRegistryModelMetadata] = None,
+        remove_metadata: Optional[List[str]] = None,
+        stage: Optional[ModelVersionStage] = None,
+    ) -> ModelVersion:
+        """Updates a model version in the model registry.
+
+        Args:
+            name: The name of the registered model.
+            version: The version of the model version to update.
+            description: The description of the model version.
+            metadata: Metadata associated with this model version.
+            remove_metadata: The metadata to remove from the model version.
+            stage: Not supported in Huggingface.
+
+        Returns:
+            The updated model version.
+
+        Raises:
+            KeyError: If the model version does not exist.
+            RuntimeError: If update fails.
+        """
+        registered_model = self.update_model(
+            name=name,
+            description=description,
+            metadata=metadata.dict() if metadata else {},
+            remove_metadata=remove_metadata,
+            revision=version,
+        )
+        pr_num = version.split("/")[-1]
+        if stage:
+            if (
+                stage != ModelVersionStage.PRODUCTION
+                and stage != ModelVersionStage.PRODUCTION.value
+            ):
+                raise RuntimeError(
+                    "Only `ModelVersionStage.PRODUCTION` stage is supported with Huggingface at the moment."
+                )
```

Comment:
Why? and wouldn't HF be used for non-prod things rather than prod stage? Would add a TODO or JIRA ticket for this. Feels a bit arbitrary?

Reply:
This is cause we can only PR versus main atm